From 043075bf719f4efd87fe968e3ec80af09b460718 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 15 Oct 2021 06:31:41 -0700
Subject: [PATCH] webpack: use sha256 instead of ancient md4 to fix build with
 openssl-3

:Release Notes:
Follow the suggestion from:
https://stackoverflow.com/questions/69394632/webpack-build-failing-with-err-ossl-evp-unsupported

:Detailed Notes:
To fix build with openssl-3:

added 721 packages from 568 contributors in 37.661s

8 packages are looking for funding
  run `npm fund` for details

npm verb exit [ 0, true ]
npm timing npm Completed in 39099ms
npm info ok

> com.webos.service.flowmanager@1.0.0 webpack /jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git
> ./node_modules/webpack/bin/webpack.js --mode=development

internal/crypto/hash.js:46
  this[kHandle] = new _Hash(algorithm, xofLen);
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (internal/crypto/hash.js:46:19)
    at Object.createHash (crypto.js:116:10)
    at module.exports (/jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/webpack/lib/NormalModule.js:471:10)
    at /jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/webpack/lib/NormalModule.js:503:5
    at /jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/webpack/lib/NormalModule.js:358:12
    at /jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at Array.<anonymous> (/jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
    at Storage.finished (/jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
    at /jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
    at /jenkins/mjansa/build/ros/webos-foxy-kirkstone/tmp-glibc/work/qemux86-webos-linux/com.webos.service.flowmanager/1.0.0-8-r6/git/node_modules/graceful-fs/graceful-fs.js:123:16
    at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:73:3) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! com.webos.service.flowmanager@1.0.0 webpack: `./node_modules/webpack/bin/webpack.js --mode=development`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the com.webos.service.flowmanager@1.0.0 webpack script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-310] Apply .patch files for hot-fix to corresponding repositories
[WRP-10791] CCC: Various build fixes

Change-Id: I39a0e0edd6ef2fac9692c8d6affffe8f3b84e66f
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/com.webos.service.flowmanager/+/348056 webpack: use sha256 instead of ancient md4 to fix build with openssl-3]

 webpack.config.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/webpack.config.js b/webpack.config.js
index de99e0f..50e54a7 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -11,6 +11,7 @@ const serverConfig = {
         app: APP_DIR + '/index.js'
     },
     output: {
+        hashFunction: "sha256",
         filename: 'index.js',
         path: BUILD_DIR
     },
